flink-connector-pulsar/src/main/java/org/apache/flink/connector/pulsar/source/PulsarSourceBuilder.java

```diff
@@ -619,7 +619,7 @@ private void ensureSubscriberIsNull(String attemptingSubscribeMode) {
 
     private void ensureSchemaTypeIsValid(Schema<?> schema) {
         SchemaInfo info = schema.getSchemaInfo();
-        if (info.getType() == SchemaType.AUTO_CONSUME || info.getType() == SchemaType.AUTO) {
+        if (info.getType() == SchemaType.AUTO_CONSUME) {
```

Comment:
I just notice this is a deprecated API calling in Pulsar and could never happen. Since this is a code refactor PR, I add it here and didn't submit a new PR.